service/frontend/workflow_handler_second_test.go

```diff
@@ -0,0 +1,139 @@
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package frontend
+
+import (
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	commonpb "go.temporal.io/api/common/v1"
+	enumspb "go.temporal.io/api/enums/v1"
+	"go.temporal.io/api/schedule/v1"
+	workflowpb "go.temporal.io/api/workflow/v1"
+
+	schedspb "go.temporal.io/server/api/schedule/v1"
+	"go.temporal.io/server/common/payload"
+	"go.temporal.io/server/common/persistence/visibility/manager"
+	"go.temporal.io/server/common/searchattribute"
+)
+
+func TestDescribeScheduleAnnotatesScheduledWorkflowWithTypes(t *testing.T) {
+	controller := gomock.NewController(t)
+	defer controller.Finish()
+	visibilityManager := makeVisibilityManagerStub(controller)
+	searchAttrProvider := makeSearchAttributesProviderStub(
+		controller,
+		map[string]enumspb.IndexedValueType{
+			"CustomKeywordField": enumspb.INDEXED_VALUE_TYPE_TEXT,
+			"CustomBoolField":    enumspb.INDEXED_VALUE_TYPE_BOOL,
+		},
+	)
+	h := makeWorkflowHandler(visibilityManager, searchAttrProvider)
+
+	response := makeResponseWithScheduledWorkflowAttributes(
+		map[string]string{"CustomKeywordField": "keyword", "CustomBoolField": "true"})
+
+	err := h.annotateSearchAttributesOfScheduledWorkflow(response, "ns")
+	if err != nil {
+		t.Fatalf("error %v", err)
+	}
+	assertScheduledWorkflowSearchAttributeHasAssociatedTypeOf(
+		t, response, "CustomKeywordField", enumspb.INDEXED_VALUE_TYPE_TEXT)
+	assertScheduledWorkflowSearchAttributeHasAssociatedTypeOf(
+		t, response, "CustomBoolField", enumspb.INDEXED_VALUE_TYPE_BOOL)
+}
+
+func assertScheduledWorkflowSearchAttributeHasAssociatedTypeOf(
+	t *testing.T,
+	response *schedspb.DescribeResponse,
+	searchAttribute string,
+	expectedType enumspb.IndexedValueType,
+) {
+	t.Helper()
+	attributes := getScheduledWorkflowSearchAttributes(response)
+	actualAttributeType := attributes.IndexedFields["AliasFor"+searchAttribute].Metadata[searchattribute.MetadataType]
+	if string(actualAttributeType) != expectedType.String() {
+		t.Errorf(
+			"expected type %s for attribute %s, got %s",
+			expectedType.String(),
+			searchAttribute,
+			string(actualAttributeType),
+		)
+	}
+}
+
+func getScheduledWorkflowSearchAttributes(response *schedspb.DescribeResponse) *commonpb.SearchAttributes {
+	return response.Schedule.Action.Action.(*schedule.ScheduleAction_StartWorkflow).StartWorkflow.SearchAttributes
+}
+
+func makeWorkflowHandler(
```

Comment:
be careful with generic names like this.. all tests in this package share the same top-level namespace